artemis-journal/src/main/java/org/apache/activemq/artemis/core/io/buffer/counters/Profiler.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.activemq.artemis.core.io.buffer.counters;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.apache.activemq.artemis.journal.ActiveMQJournalLogger;
+
+/**
+ * Factory class to instantiate {@link FlushProfiler}s classes.
+ */
+public final class Profiler {
+
+   private static final FlushProfiler BLACK_HOLE_PROFILER = new FlushProfiler() {
```

Comment:
The ``default`` method idea is good IMHO when you've methods not bounded each others, but when you have methods that makes sense to be implemented (and called) together (eg Lock::lock and Lock::unlock), I prefer to use normal methods and force the implementation to have both.